pkg/destroy/alibabacloud/alibabacloud.go

```diff
@@ -1001,32 +1011,33 @@ func (o *ClusterUninstaller) deleteEcsInstances(logger logrus.FieldLogger) (err
 		instanceIDs = append(instanceIDs, instanceArn.ResourceID)
 	}
 
-	err = o.modifyECSInstancesDeletionProtection(instanceIDs, logger)
-	if err != nil {
-		return err
-	}
-
-	logger.WithField("ecsIDs", instanceIDs).Debug("Deleting ECS instances")
-	request := ecs.CreateDeleteInstancesRequest()
-	request.InstanceId = &instanceIDs
-	request.Force = "true"
-	_, err = o.ecsClient.DeleteInstances(request)
-	if err != nil {
-		return err
+	// The paramater InstanceIds of DeleteInstances and DescribeInstances consists of up to 100 instance IDs
+	maxInstanceCount := 100
+	for _, ids := range splitArray(instanceIDs, maxInstanceCount) {
+		err = o.modifyECSInstancesDeletionProtection(ids, logger)
+		if err != nil {
+			return err
+		}
+		err = o.deleteEcsInstance(ids, logger)
+		if err != nil {
+			return err
+		}
 	}
 
 	err = wait.Poll(
 		5*time.Second,
 		5*time.Minute,
 		func() (bool, error) {
-			response, err := o.listEcsInstance(instanceIDs)
-			if err != nil {
-				return false, err
-			}
-			if response.TotalCount == 0 {
-				return true, nil
+			for _, ids := range splitArray(instanceIDs, maxInstanceCount) {
```

Comment:
we are calling `splitArray` twice from within the same function. Here and https://github.com/openshift/installer/pull/5659/files#diff-5d31e7ab73dc252cc09331a0790fcc8b6cd3add977402effbe19a1b1b24fba39R1016. 

Can't we just call it once?